### PR TITLE
Add retrieveManagedSync

### PR DESCRIPTION
**Pull request by ajsquared**
Adds retrieveManagedSync key that, when set to true, enables synchronizing retrieved to the current build by removed unneeded files.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by ajsquared** [comment]
Pushed a fix for the test that did not compile

**Comment by ajsquared** [comment]
I'm trying to reproduce the failures from CI, but am having difficulty doing so.

File: ivy/src/main/scala/sbt/IvyActions.scala [hunk 1]
```diff
@@ -31,7 +31,7 @@ final class UpdateConfiguration(val retrieve: Option[RetrieveConfiguration], val
     logging: UpdateLogging.Value = this.logging): UpdateConfiguration =
     new UpdateConfiguration(retrieve, missingOk, logging)
 }
-final class RetrieveConfiguration(val retrieveDirectory: File, val outputPattern: String)
+final class RetrieveConfiguration(val retrieveDirectory: File, val outputPattern: String, val sync: Boolean)
```
Comment (jsuereth) [inline ivy/src/main/scala/sbt/IvyActions.scala]: Given this is a binary incompatible change, you'll ahve to add a new constructor with the old patterns and instantiates a default sync value for boolean.

Comment (ajsquared) [inline ivy/src/main/scala/sbt/IvyActions.scala]: Okay, I will add that in.

**Comment by eed3si9n** [comment]
I've been restarting Travis CI jobs to clear out the transient failures. There's still one more pending but we should get green soon.
Otherwise LGTM.

**Comment by jsuereth** [comment]
Also, looks good to me.

**Comment by ajsquared** [comment]
@eed3si9n It looks like the `dependency-management/eviction-warning` scripted test failed again in `mavenResolverPluginTest:scripted dependency-management/*1of2 project/transitive-plugins`.  I'm not able to reproduce this failure.

**Pull request merged**